helm/aws-admission-controller/templates/webhook.yaml

```diff
@@ -8,6 +8,25 @@ metadata:
   labels:
     {{- include "labels.common" . | nindent 4 }}
 webhooks:
+  - name: awsclusters.{{ include "resource.default.name" . }}.giantswarm.io
+    admissionReviewVersions: [v1]
+    failurePolicy: Ignore
+    sideEffects: None
+    clientConfig:
+      service:
+        name: {{ include "resource.default.name" . }}
+        namespace: {{ include "resource.default.namespace" . }}
+        path: /mutate/awscluster
+      caBundle: Cg==
+    rules:
+      - apiGroups: ["infrastructure.giantswarm.io"]
+        resources:
+          - awsclusters
+        apiVersions:
+          - v1alpha2
+        operations:
+          - CREATE
+          - UPDATE
```

Comment:
Do we care about UPDATE operations or shall this only be checked when someone creates the CR?

The reason why I'm asking is because we do not verify the `mutator.Deserializer.Decode(request.OldObject.Raw, nil, awsCluster)`

Reply:
I could imagine it is prudent to default on creation but verify that nothing changes anymore upon updates. IIUTC once set the CIDR must not change. Depending on the implementation in the webhook we might just want to register separate defaulters and validators.

Reply:
I can leave this out of the PR for now, but we should generally think about how we want to handle the updates and creations because we have a lot of different cases. As it is right now we don't really need the UPDATE for the AWSCluster but we might in the future. We also do not have validation for it but we might also want to add that. (For example, ensure that the CIDR isn't changed as you suggest)
In some other CRs we specifically perform certain defaulting on update and validation on creation and vice versa already. We also register separate defaulting and validating webhooks already but we only distinguish between different operations inside the admission process. I would like to talk about this as a team.